File: parallel/hellompi.c

#include <stdio.h>
#include <mpi.h>

int main(int argc, char *argv[]) {
    int ierr, rank;
    ierr = MPI_Init(&argc, &argv);

    ierr = MPI_Comm_rank(MPI_COMM_WORLD, &rank);
    printf("C: Hello World from %i!\n", rank);

    ierr = MPI_Finalize();
    return 0;
}

File: parallel/hellompi.f

      program hello
      include 'mpif.h'
      integer ierr, rank

      call MPI_INIT(ierr)
      call MPI_COMM_RANK(MPI_COMM_WORLD, rank, ierr)

      write(*,'(A,I1,A)') "F: Hello World from ", rank, "!"

      call MPI_FINALIZE(ierr)
      end program
